### PR TITLE
chore: merge traces into a single zip file

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: packages/trace-viewer/src/traceModel.ts [hunk 1]
```diff
@@ -47,50 +46,54 @@ export class TraceModel {
...
-    if (stacksEntry) {
-      const writer = new zipjs.TextWriter();
-      await stacksEntry.getData!(writer);
-      const metadataMap = parseClientSideCallMetadata(JSON.parse(await writer.getData()));
-      for (const action of this.contextEntry.actions)
-        action.stack = action.stack || metadataMap.get(action.callId);
-    }
+      const stacksWriter = new zipjs.TextWriter();
+      const stacksEntry = this._entries.get(ordinal + 'trace.stacks')!;
+      if (stacksEntry) {
```
Comment (dgozman) [inline packages/trace-viewer/src/traceModel.ts:87]: nit: either null-check here or question mark on the previous line, but not both?

File: tests/playwright-test/reporter-html.spec.ts [hunk 2]
```diff
@@ -443,12 +443,12 @@ test('should show multi trace source', async ({ runInlineTest, page, server, sho
...
   // Expect one image-link to trace viewer and 2 separate download links
   await expect(page.locator('img')).toHaveCount(1);
-  await expect(page.locator('a', { hasText: 'trace' })).toHaveText(['trace-1', 'trace-2']);
+  await expect(page.locator('a', { hasText: 'trace' })).toHaveText(['trace']);
 
   await page.click('img');
   await page.click('.action-title >> text=page.evaluate');
   await page.click('text=Source');
-  await expect(page.locator('.source-line-running')).toContainText('page.evaluate');
+  await expect(page.locator('.source-line-running')).toContainText('page.evaluate', { timeout: 0 });
```
Comment (dgozman) [inline tests/playwright-test/reporter-html.spec.ts:451]: Why zero here?

File: tests/config/utils.ts [hunk 3]
```diff
@@ -102,17 +103,30 @@ export async function parseTrace(file: string): Promise<{ events: any[], resourc
...
   }
 
-  const stacks = parseClientSideCallMetadata(JSON.parse(resources.get('trace.stacks')!.toString()));
+  const stacks: Map<string, StackFrame[]> = new Map();
+  for (const stacksFile of [...resources.keys()].filter(name => name.endsWith('.stacks'))) {
+    for (const [key, value] of parseClientSideCallMetadata(JSON.parse(resources.get(stacksFile).toString())))
+      stacks.set(key, value);
+  }
+
+  const metadatas: SerializedClientSideCallMetadata[]  = [];
```
Comment (dgozman) [inline tests/config/utils.ts:126]: `metadatas` look unused.

File: packages/trace-viewer/src/traceModel.ts [hunk 4]
```diff
@@ -116,74 +119,70 @@ export class TraceModel {
     return this._snapshotStorage!;
   }
 
-  private _build() {
-    this.contextEntry!.actions.sort((a1, a2) => a1.startTime - a2.startTime);
-    this.contextEntry!.resources = this._snapshotStorage!.resources();
```
Comment (dgozman) [inline packages/trace-viewer/src/traceModel.ts:121]: We never set `resources` anymore, perhaps remove them from the type definition?

**Review by dgozman — APPROVED** [review]

File: packages/trace-viewer/src/traceModel.ts (hunk 4 above)
Comment (pavelfeldman) [reply packages/trace-viewer/src/traceModel.ts:121]: We do populate it incrementally: contextEntry.resources.push(event.snapshot);

**Pull request merged**